src/Parser/Mage/Frost/Modules/Features/GlacialSpike.js

```diff
@@ -6,48 +6,57 @@ import SpellIcon from 'common/SpellIcon';
 import SpellLink from 'common/SpellLink';
 import { formatPercentage } from 'common/format';
 import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+import EnemyInstances, { encodeTargetString } from 'Parser/Core/Modules/EnemyInstances';
+import AbilityTracker from 'Parser/Core/Modules/AbilityTracker';
 
 class GlacialSpike extends Analyzer {
-  overcapped = 0;
-  total = 0;
+  static dependencies = {
+    enemies: EnemyInstances,
+    abilityTracker: AbilityTracker,
+  };
 
   constructor(...args) {
     super(...args);
     this.active = this.selectedCombatant.hasTalent(SPELLS.GLACIAL_SPIKE_TALENT.id);
+    this.hasEbonbolt = this.selectedCombatant.hasTalent(SPELLS.EBONBOLT_TALENT.id);
   }
 
-  on_toPlayer_changebuffstack(event) {
+  badCasts = 0
+
+  on_byPlayer_damage(event) {
     const spellId = event.ability.guid;
-    if (spellId === SPELLS.ICICLES_BUFF.id && event.newStacks > event.oldStacks) {
-      this.total += 1;
+    if (spellId !== SPELLS.GLACIAL_SPIKE_DAMAGE.id) {
+      return;
     }
-  }
 
-  on_toPlayer_refreshbuff(event) {
-    if (event.ability.guid === SPELLS.GLACIAL_SPIKE_BUFF.id) {
-      this.overcapped += 1;
+    const damageTarget = encodeTargetString(event.targetID, event.targetInstance);
+    const enemy = this.enemies.getEntity(event);
+    if (this.castTarget === damageTarget && !enemy.hasBuff(SPELLS.WINTERS_CHILL.id)) {
```

Comment:
Doesn't Frost mage check for winter's chill on cast and not on damage?

Reply:
Winters chill has a very short duration on the target, and isn't applied until the brain freeze flurry hits the target. So when the glacial spike is cast, winters chill won't be on the target cause flurry hasn't been cast yet, but flurry travels faster than glacial spike so it will land before glacial spike and thus gs will land into winters chill (as long as they aren't standing on top of the boss)

Reply:
I'm pretty sure it actually does check on damage